### ISSUE TITLE
NSURLSession redirect 301 Location not work

### ISSUE THREAD

**Issue by onmyway133** [Alamofire/Alamofire#1165]
I have a request that returns 301 with header `"Location": "another_url_link"` but my `response` block get called without redirection :cry: 

```swift
request
        .validate()
        .response { request, response, data, error in
   // response contains status code 301 with header containing "Location"
}
```

I configure @cnoon  https://github.com/Alamofire/Alamofire/pull/317 `taskWillPerformHTTPRedirectionWithCompletion` but it does not get called.

**Comment by cnoon** [comment]
There was a bug that was fixed in the `3.3.1` that was causing the `taskWillPerformHTTPRedirectionWithCompletion` closure to not be called @onmyway133. If you update to the latest release, everything should work as expected.

I'm going to close this issue out for now. If this doesn't actually solve your problem, please provide some additional info and I'll be happy to re-open.

Cheers. 🍻

**Comment by luohui8891** [comment]
@cnoon Can you try the following URL it didn't work with latest Alamofire.
`http://www.hkpro.com/forum/attachment.php?attachmentid=104674`
